File: src/main/golang/leetcode/editor/cn/[191]Number of 1 Bits.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func hammingWeight(num uint32) (ans int) {
	for num != 0 {
		num &= (num - 1)
		ans += 1
	}
	return
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[191]Number of 1 Bits.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def hammingWeight(self, n: int) -> int:
        ans = 0
        while n:
            ans += 1
            n &= n - 1
        return ans

# leetcode submit region end(Prohibit modification and deletion)
